llama-index-core/llama_index/core/workflow/context.py

```diff
@@ -49,6 +56,85 @@ def __init__(self, workflow: "Workflow", stepwise: bool = False) -> None:
         # Step-specific instance
         self._events_buffer: Dict[Type[Event], List[Event]] = defaultdict(list)
 
+    def _serialize_queue(self, queue: asyncio.Queue, serializer: BaseSerializer) -> str:
+        queue_items = list(queue._queue)  # type: ignore
+        queue_objs = [serializer.serialize(obj) for obj in queue_items]
+        return json.dumps(queue_objs)  # type: ignore
+
+    def _deserialize_queue(
+        self, queue_str: str, serializer: BaseSerializer
+    ) -> asyncio.Queue:
+        queue_objs = json.loads(queue_str)
+        queue = asyncio.Queue()  # type: ignore
+        for obj in queue_objs:
+            event_obj = serializer.deserialize(obj)
+            queue.put_nowait(event_obj)
+        return queue
+
+    def _serialize_globals(self, serializer: BaseSerializer) -> Dict[str, Any]:
+        serialized_globals = {}
+        for key, value in self._globals.items():
+            try:
+                serialized_globals[key] = serializer.serialize(value)
+            except Exception as e:
+                raise ValueError(f"Failed to serialize value for key {key}: {e}")
+        return serialized_globals
+
+    def _deserialize_globals(
+        self, serialized_globals: Dict[str, Any], serializer: BaseSerializer
+    ) -> Dict[str, Any]:
+        deserialized_globals = {}
+        for key, value in serialized_globals.items():
+            try:
+                deserialized_globals[key] = serializer.deserialize(value)
+            except Exception as e:
+                raise ValueError(f"Failed to deserialize value for key {key}: {e}")
+        return deserialized_globals
+
+    def to_dict(self, serializer: Optional[BaseSerializer] = None) -> Dict[str, Any]:
```

Comment:
minor nit: wondering if it would be useful to make this into at least a `TypedDict`?